mac/Keyman4MacIM/Keyman4MacIM/KMInfoWindow/KMInfoWindowController.m

```diff
@@ -124,31 +125,24 @@ - (NSString *)detailsHtml {
         "</html>";
         
         NSString *pBodyFormat = @"<p class='body'>%@</p><br>";
-        NSMutableString *kbsStr = [NSMutableString stringWithString:@""];
+        NSMutableString *keyboardString = [NSMutableString stringWithString:@""];
         
-        // TODO: getting keyboards from folder/kmx rather than inf
-        NSArray *kbs = [[self AppDelegate] keyboardNamesFromFolder:self.packagePath];
-        if (kbs != nil && kbs.count) {
-            for (NSArray *kbName in kbs) {
-                [kbsStr appendString:[NSString stringWithFormat:pBodyFormat, kbName]];
+        if (self.packageInfo.keyboards.count) {
+            for (KMKeyboardInfo *keyboard in self.packageInfo.keyboards) {
+                [keyboardString appendString:[NSString stringWithFormat:pBodyFormat, keyboard.name]];
             }
-        }
-        // TODO: need to provide above as fallback in case no json???
-/*        else {
-            // This was the old logic (prior to fixing issue #994). Keeping as fallback, though
-            // it's unlikely to be useful/needed.
-            kbs = [self.keyboardInfo objectForKey:kKeyboard];
-        
-            if (kbs != nil && kbs.count) {
-                for (NSArray *kb in kbs) {
-                    [kbsStr appendString:[NSString stringWithFormat:pBodyFormat, [kb objectAtIndex:0]]];
+        } else {
+            // get keyboards from folder because not available in package info (.json or .inf)
```

Comment:
In some ways this is a nice fallback but it's going above and beyond what Keyman for Windows does -- we should probably treat a package with no keyboard metadata as borken rather than trying to get something out of it (principle: inconsistency between platforms. So I think we don't need to do this folder scan at all -- even though that's what Keyman for Mac did previously.

Reply:
ok, I'll remove that and just have an empty keyboard list if none are found in the metadata